osu.Game/Screens/Edit/Screens/Setup/Components/LabelledComponents/LabelledSwitchButton.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework.Allocation;
+using osu.Framework.Configuration;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.UserInterface;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+
+namespace osu.Game.Screens.Edit.Screens.Setup.Components.LabelledComponents
+{
+    public class LabelledSwitchButton : CompositeDrawable, IHasCurrentValue<bool>
+    {
+        private readonly Box background;
+        private readonly OsuSpriteText label;
+        private readonly OsuSpriteText bottomText;
+
+        private const float corner_radius = 15;
+        private const float default_label_text_size = 16;
+        private const float default_bottom_label_text_size = 12;
+        private const float normal_height = 40;
+        private const float default_label_horizontal_offset = 15;
+        private const float default_label_vertical_offset = 12;
+        private const float default_switch_horizontal_offset = 15;
+        private const float default_switch_vertical_offset = 10;
+
+        public Bindable<bool> Current { get; } = new Bindable<bool>();
+
+        public string LabelText
+        {
+            get => label.Text;
+            set => label.Text = value;
+        }
+
+        public string BottomLabelText
+        {
+            get => bottomText.Text;
+            set
+            {
+                bottomText.Text = value;
+                Height = normal_height + (value != "" ? 20 : 0);
+            }
+        }
+
+        public float LabelTextSize
+        {
+            get => label.TextSize;
+            set => label.TextSize = value;
+        }
+
+        public Color4 LabelTextColour
+        {
+            get => label.Colour;
+            set => label.Colour = value;
+        }
+
+        public Color4 BackgroundColour
+        {
+            get => background.Colour;
+            set => background.Colour = value;
+        }
+
+        public LabelledSwitchButton()
+        {
+            SwitchButton switchButton;
+
+            RelativeSizeAxes = Axes.X;
+            Height = normal_height;
+            Masking = true;
+            CornerRadius = corner_radius;
+
+            InternalChild = new Container
```

Comment:
is this container even doing anything?

Reply:
It's necessary, otherwise the masking fails due to the applied padding to the component, if I'm not mistaken.

Reply:
Sounds like an incorrect use of padding, possibly? Did you want margin instead?

Reply:
In the previous implementations, @smoogipoo approved the construction of the components like this, and the usage of paddings was declared to be irrelevant to the component itself. The conclusion is, by testing, the container is needed.

Reply:
Could you point me to where the padding is applied? I can't even find it.

Reply:
It's applied on the container that contains the component in the test case, turns out removing the extra container does not break it, it's going to be removed.